collections/mod.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+/*
+ * Stack is a first-in-last-out data structure.
+ *
+ * Create by optionally passing an Iterable:
+ *
+ *      const s = Stack([1, 2, 3])
+ *      console.log(s.pop())  // 3
+ *
+ */
+export class Stack<T> implements Iterable<T> {
+  private _array: T[];
+  constructor(ls?: Iterable<T>) {
+    this.clear();
+    for (const item of ls || []) {
+      this._array.push(item);
+    }
+  }
+  clear(): void {
+    this._array = [];
+  }
+  peek(): T {
+    return this._array[this._array.length - 1];
+  }
+  pop(): T {
+    return this._array.pop();
+  }
+  push(value): void {
+    this._array.push(value);
+  }
+  size(): number {
+    return this._array.length;
+  }
+
+  [Symbol.iterator]() {
```

Comment:
Can you not just forward to the array's iterator function?

Reply:
Well, one reverses and one starts at offset. Perhaps there's a better way?